File: JAVA/academic_people/Professor.java
package pessoa_academico;
public class Professor extends Person{
    private String employeeId;
    private double salary;

    public Professor(){}
    public Professor(String name){
        setName(name);
    }

    public String getEmployeeId() {
        return employeeId;
    }
    public void setEmployeeId(String employeeId) {
        this.employeeId = employeeId;
    }
    public double getSalary() {
        return salary;
    }
    public void setSalary(double salary) {
        this.salary = salary;
    }
    public void correctTest(Student aluno){
        if(aluno.getGrade()>6){
            System.out.println("Student passed exam");
            aluno.setDependence(true);
        }
        else{
            System.out.println("Student didnt pass exam :(");
            aluno.setDependence(false);
        }
    }
}

File: CPP/pessoa_academico/Professor.cpp
#include <iostream>
using namespace std;
#include "header.h"

class Professor : public Person{
    private:
    string employeeId;
    double salary;

    public:
    Professor(){}
    Professor(string name){
        setName(name);
    }

    string getEmployeeId() {
        return employeeId;
    }
    void setEmployeeId(string _employeeId) {
        employeeId = employeeId;
    }
    double getSalary() {
        return salary;
    }
    void setSalary(double _salary) {
        salary = _salary;
    }
    void correctTest(Student aluno){
        if(aluno.getGrade()>6){
            cout << "Student passed exam" << endl;
            aluno.setDependence(true);
        }
        else{
            cout << "Student didnt pass exam :(" << endl;
            aluno.setDependence(false);
        }
    }
};
